src/vmm/src/devices/virtio/queue.rs

```diff
@@ -449,15 +449,19 @@ impl Queue {
             len,
         };
         self.write_used_ring(mem, next_used, used_element)?;
+        self.advance_used_ring(mem, 1);
+        Ok(())
+    }
 
-        self.num_added += Wrapping(1);
-        self.next_used += Wrapping(1);
+    /// Advance number of used descriptor heads by `n`.
+    pub fn advance_used_ring<M: GuestMemory>(&mut self, mem: &M, n: u16) {
+        self.num_added += Wrapping(n);
+        self.next_used += Wrapping(n);
```

Comment:
The confusing bit is that suddenly there is a method that allows to "forward" the used ring by more than one descriptors at the time. This is not used anywhere in the code (in the sense that only `advance_used_ring(mem, 1);` is ever called). I assume you want to do things with many descriptors (write memory in them) and then once everything is done, do a batch of `add_used`. It makes sense, but in the context of this PR is confusing because nothing is using it. Please mention what this thing does and why it will be needed in the future.

Again, it might be obvious to you *now* why this is useful but to someone who is not accustomed with what you are trying to do and will be trying to debug something 6 months from now, this commit will be cryptic. So, right details about what did you do and why in the commit message.

Reply:
Removed commit from this PR.